Transformations/Models.MetaTransformation/Meta/Class2Type.cs

```diff
@@ -1050,12 +1046,12 @@ protected virtual CodeMemberMethod CreateGetCompositionName(IClass input, CodeTy
                 };
                 getCompositionName.Parameters.Add(new CodeParameterDeclarationExpression(typeof(object), "container"));
                 var containerRef = new CodeArgumentReferenceExpression("container");
-                AddReferencesOfClass(input, generatedType, (stmts, r, ctx) =>
+                AddReferencesOfClass(input, generatedType, (stmts, r, p, ctx) =>
                 {
                     if (r.UpperBound != 1)
                     {
-                        stmts.Add(new CodeConditionStatement(new CodeBinaryOperatorExpression(containerRef, CodeBinaryOperatorType.ValueEquality, new CodeFieldReferenceExpression(new CodeThisReferenceExpression(), "_" + r.Name.ToCamelCase())),
-                            new CodeMethodReturnStatement(new CodePrimitiveExpression(r.Name))));
+                        stmts.Add(new CodeConditionStatement(new CodeBinaryOperatorExpression(containerRef, CodeBinaryOperatorType.ValueEquality, new CodeFieldReferenceExpression(new CodeThisReferenceExpression(), "_" + p.Name.ToCamelCase())),
+                            new CodeMethodReturnStatement(new CodePrimitiveExpression(p.Name))));
```

Comment:
No, this should really return the reference name in the metamodel space, not in the code space. Therefore, please revert to r.Name. The change of line 1057, however, is correct.

Reply:
Thanks Georg.  I've reverted that change.